src/components/Choice/src/Choice.vue

```diff
@@ -18,10 +19,35 @@ import key from './key';
 
 export default {
 	provide() {
+		/**
+		 * Unfortunately, it's difficult to pass reactive data via a provider in Vue 2
+		 * Vue 3 will bring the `computed` helper to make (most of) this cleaner
+		 */
 		return {
 			[key]: {
-				...this.$data,
-				disabled: this.disabled,
+				isSelected: (value) => {
```

Comment:
I think we could likely simplify all this by referencing another similar component we've written in the past: https://github.com/square/orbit/blob/master/src/components/ButtonGroup/src/ButtonGroup.vue

All the data should be passed down from the parent and the child elements can leverage that data or call a method to update it.

Reply:
Fair enough!